ui/components/src/Legend/ListLegendItem.tsx

```diff
@@ -33,7 +33,6 @@ export const ListLegendItem = React.memo(function ListLegendItem({ item, sx, ...
         sx
       )}
       dense={true}
-      data-testid={`legend-item-${item.color}`}
```

Comment:
I was struggling a little with making sense of these with the colors, so I took a stab at a helper to look up the legend items by the name inside them. Let me know if you dislike the new approach. I'm not super attached to it.